kubernetes/template/DeploymentTemplates/script.sh

```diff
@@ -98,234 +137,325 @@ check_and_move_azurestack_configuration() {
-
+    
+    log_level -i "Copy ca-cert from '$AZURESTACK_ROOT_CERTIFICATE_SOURCE_PATH' to '$AZURESTACK_ROOT_CERTIFICATE_DEST_PATH' "
+    cp $AZURESTACK_ROOT_CERTIFICATE_SOURCE_PATH $AZURESTACK_ROOT_CERTIFICATE_DEST_PATH
+    
+    AZURESTACK_ROOT_CERTIFICATE_SOURCE_FINGERPRINT=`openssl x509 -in $AZURESTACK_ROOT_CERTIFICATE_SOURCE_PATH -noout -fingerprint`
     log_level -i "AZURESTACK_ROOT_CERTIFICATE_SOURCE_FINGERPRINT: $AZURESTACK_ROOT_CERTIFICATE_SOURCE_FINGERPRINT"
+    
+    AZURESTACK_ROOT_CERTIFICATE_DEST_FINGERPRINT=`openssl x509 -in $AZURESTACK_ROOT_CERTIFICATE_DEST_PATH -noout -fingerprint`
     log_level -i "AZURESTACK_ROOT_CERTIFICATE_DEST_FINGERPRINT: $AZURESTACK_ROOT_CERTIFICATE_DEST_FINGERPRINT"
-
+    
     update-ca-certificates
-
-    # Azure CLI specific changes.
+    
+    # Required by Azure CLI
     export REQUESTS_CA_BUNDLE=/etc/ssl/certs/ca-certificates.crt
-    sudo sed -i -e "\$aREQUESTS_CA_BUNDLE=/etc/ssl/certs/ca-certificates.crt" /etc/environment
-       
-    AZURESTACK_RESOURCE_METADATA_ENDPOINT="$TENANT_ENDPOINT/metadata/endpoints?api-version=2015-01-01"
-    curl $AZURESTACK_RESOURCE_METADATA_ENDPOINT
+    echo "REQUESTS_CA_BUNDLE=/etc/ssl/certs/ca-certificates.crt" | tee -a /etc/environment > /dev/null
+    
+    if [ $IDENTITY_SYSTEM == "ADFS" ]; then
+        # Trim "adfs" suffix
+        ENDPOINT_ACTIVE_DIRECTORY_ENDPOINT=`echo $METADATA | jq '.authentication.loginEndpoint' | xargs | sed -e 's/adfs*$//' | xargs`
+        log_level -i "ENDPOINT_ACTIVE_DIRECTORY_ENDPOINT: $ENDPOINT_ACTIVE_DIRECTORY_ENDPOINT"
+        
+        CERTIFICATE_PFX_LOCATION="spnauth.pfx"
+        CERTIFICATE_PEM_LOCATION="spnauth.pem"
+        
+        convert_secret_to_cert $SPN_CLIENT_SECRET $CERTIFICATE_PFX_LOCATION $CERTIFICATE_PEM_LOCATION
+        
+        log_level -i "PFX path: '$CERTIFICATE_PFX_LOCATION'"
+        log_level -i "PEM path: '$CERTIFICATE_PEM_LOCATION'"
+    fi
+}
+
+# Add azure-cli apt source
+add_azurecli_source()
+{
+    # https://docs.microsoft.com/en-us/azure/azure-stack/user/azure-stack-version-profiles-azurecli2#connect-to-azure-stack
+    # https://docs.microsoft.com/en-us/cli/azure/install-azure-cli-apt?view=azure-cli-latest
+    
+    log_level -i "Adding azure-cli apt source."
+    AZ_REPO=$(lsb_release -cs)
+    echo "deb [arch=amd64] https://packages.microsoft.com/repos/azure-cli/ $AZ_REPO main" | tee /etc/apt/sources.list.d/azure-cli.list
+    
+    log_level -i "Downloading the Microsoft signing key."
+    RECV_KEY=BC528686B50D79E339D3721CEB3E94ADBE1229CF
+    retrycmd_if_failure 5 10 60 apt-key \
+    --keyring /etc/apt/trusted.gpg.d/Microsoft.gpg adv \
+    --keyserver packages.microsoft.com \
+    --recv-keys $RECV_KEY \
+    || exit $ERR_MS_GPG_KEY_DOWNLOAD_TIMEOUT
+}
+
+# Clone msazurestackworkloads' AKSe fork and move relevant files to the working directory
+download_akse()
+{
+    # TODO REMOVE
+    rm -rf aks-engine/ bin/
```

Comment:
I think we should either remove this or pipe the output out as in fresh deployment it will always fail.

Reply:
Yeah, that line was meant to help troubleshooting, will delete. 
FYI, it won't cause failures on fresh deployments